File: lib/typescript/id.ts
import { Resolver } from 'did-resolver';
import fetch from 'node-fetch';

async function resolveDid(didUrl: string, resolverUrl: string) {
  const resolver = new Resolver({ fetch });
  try {
    const didDocument = await resolver.resolve(didUrl);
    console.log('Resolved DID Document:', didDocument);
  } catch (error) {
    console.error('Failed to resolve DID:', error);
  }
}

const didUrl = 'did:example:123456789abcdefghi';
const resolverUrl = 'https://example-did-resolver.com/resolve';
resolveDid(didUrl, resolverUrl);

File: lib/rust/id.rs
use reqwest::blocking::get;
use reqwest::Error;

fn resolve_did(did_url: &str, resolver_url: &str) -> Result<String, Error> {
    let url = format!("{}?did={}", resolver_url, did_url);
    let response = get(&url)?;
    
    match response.status().is_success() {
        true => Ok(response.text()?),
        false => Err(Error::new(reqwest::StatusCode::INTERNAL_SERVER_ERROR, "Failed to resolve DID")),
    }
}

fn main() {
    let did_url = "did:example:123456789abcdefghi";
    let resolver_url = "https://example-did-resolver.com/resolve";

    match resolve_did(did_url, resolver_url) {
        Ok(did_document) => println!("Resolved DID Document: {}", did_document),
        Err(err) => eprintln!("Error: {}", err),
    }
}

